### ISSUE TITLE
Any interest in serde support?

### ISSUE THREAD

**Issue by dgunay** [greyblake/ta-rs#28]
A while back I forked this repo for a summer project and one of the things I added to my fork was serde support before I got busy with school and work. If I opened a PR for that would it be welcome? In its present state it may be a bit disruptive to the API since if I recall correctly I had to adjust one of the traits.

**Comment by greyblake** [comment]
Hi @dgunay,
ta is not involved in any kind of data serialization.
Could you please clarify what do exactly mean by serde support?

**Comment by dgunay** [comment]
Just submitted a PR - basically, it allows the user to choose any serialization crate that is compatible with [serde](serde.rs) (JSON, binary, etc) and then use it to serialize/deserialize the indicators. For my use case, I was serializing machine learning models (which contained various `ta::indicators`) to disk using a binary format. 

I added [derived Serialize/Deserialize](https://serde.rs/derive.html) to the technical indicators to do it automatically.

While it is possible for the user to implement this themselves for remote types, it is an awkward process that requires the user to essentially [redefine the struct's shape in their own crate](https://serde.rs/remote-derive.html). It is a brittle solution that breaks if the upstream code changes the struct definition in any way. 

I would have had to do this for every indicator I wanted to use, but I wanted forward compatibility with any `ta::indicators::*` so I forked the repository and derived Serialize/Deserialize to accomplish this.
